File: python/day02/day02.py
from pathlib import Path


def get_file_contents() -> list:
    path = Path(__file__).parent / "../resources/day02.txt"
    with path.open() as f:
        lines = f.readlines()

    return lines


def solve_part1(data: list) -> int:
    horiztonal = 0
    depth = 0

    for line in data:
        direction, quantity = line.split(' ')
        quantity = int(quantity)

        if 'forward' == direction:
            horiztonal = horiztonal + quantity
        elif 'up' == direction:
            depth = depth - quantity
        else:
            depth = depth + quantity

    return horiztonal * depth


def solve_part2(data: list) -> int:
    horiztonal = 0
    depth = 0
    aim = 0

    for line in data:
        direction, amount = line.split(' ')
        amount = int(amount)

        if 'forward' == direction:
            horiztonal = horiztonal + amount
            depth = depth + (amount * aim)
        elif 'up' == direction:
            aim = aim - amount
        else:
            aim = aim + amount

    return horiztonal * depth


if __name__ == '__main__':
    data = get_file_contents()

    part1 = solve_part1(data)
    print(f'Part1: {part1}')

    part2 = solve_part2(data)
    print(f'Part 2: {part2}')

File: java/adventofcode/src/main/java/com/github/mmacdonnacha/day02/Day02.java
package com.github.mmacdonnacha.day02;

import com.github.mmacdonnacha.util.Util;
import java.util.List;

public class Day02 {

    public static void main(String[] args) {
        Util util = new Util();
        Day02 day = new Day02();
        String fileName = "day02.txt";

        List<String> data = util.getInputAsList(fileName);


        int part1 = day.solvePart1(data);
        System.out.println("Part 1: " + part1);

        // int part2 = day.solvePart2(data);
        // System.out.println("Part 2: " + part2);
    }


    public int solvePart1(List<String> data){
        int[] posAndDepth = new int[]{0, 0};
        
        for(String s : data){
            String direction = s.split(" ")[0];
            int quantity = Integer.parseInt(s.split(" ")[1]);

            switch(direction){
                case "forward":
                    posAndDepth[0] += quantity;
                    break;
                case "down":
                    posAndDepth[1] += quantity;
                    break;
                case "up":
                    posAndDepth[1] -= quantity;
                    break;
                default:
            }
        }

        return posAndDepth[0] * posAndDepth[1];
    }

    public int solvePart2(List<String> data){
        int[] posAndDepth = new int[]{0, 0};
        int aim = 0;
        
        for(String s : data){
            String direction = s.split(" ")[0];
            int qty = Integer.parseInt(s.split(" ")[1]);

            switch(direction){
                case "forward":
                    posAndDepth[0] += qty;
                    posAndDepth[1] += aim * qty;
                    break;
                case "down":
                    aim += qty;
                    break;
                case "up":
                    aim -= qty;
                    break;
                default:
            }
        }

        return posAndDepth[0] * posAndDepth[1];
    }
}
